src/solver/src/mesh_builder.C

```diff
@@ -58,144 +61,340 @@ namespace GRINS
+
+        if( input.have_variable("mesh-options/mesh_filename") )
+          {
+            std::string warning = "WARNING: mesh-options/mesh_filename is DEPRECATED.\n";
+            warning += "         Please update to use Mesh/Read/filename.\n";
+            grins_warning(warning);
+
+            mesh_filename = input("mesh-options/mesh_filename", "DIE!");
+          }
 
-    if(mesh_option=="read_mesh_from_file")
-      {
 	// According to Roy Stogner, the only read format
 	// that won't properly reset the dimension is gmsh.
 	/*! \todo Need to a check a GMSH meshes */
 	mesh->read(mesh_filename);
       }
 
-    else if(mesh_option=="create_1D_mesh")
+    // Generate the mesh using built-in libMesh functions
+    else if(mesh_build_type=="create_1D_mesh" /* This is deprecated */ ||
+            mesh_build_type=="create_2D_mesh" /* This is deprecated */ ||
+            mesh_build_type=="create_3D_mesh" /* This is deprecated */ ||
+            mesh_build_type=="generate")
+      {
+        this->generate_mesh(mesh_build_type,input,mesh);
+      }
+
+    else
+      {
+        // Shouldn't have gotten here
+        libmesh_error();
+      }
+
+    /* Only do the mesh refinement here if we don't have a restart file.
+       Otherwise, we need to wait until we've read in the restart file.
+       That is done in Simulation::check_for_restart */
+    if( !input.have_variable("restart-options/restart_file") )
+      {
+        this->do_mesh_refinement_from_input( input, comm, *mesh );
+      }
+
+    return std::tr1::shared_ptr<libMesh::UnstructuredMesh>(mesh);
+  }
+
+  void MeshBuilder::generate_mesh( const std::string& mesh_build_type, const GetPot& input,
+                                   libMesh::UnstructuredMesh* mesh )
+  {
+    unsigned int dimension = input("Mesh/Generation/dimension",0);
+
+    if( !input.have_variable("mesh-options/mesh_option") /* Deprecated */ &&
+        !input.have_variable("Mesh/Generation/dimension") )
+      {
+        libmesh_error_msg("ERROR: Must specify Mesh/Generation/dimension for generating mesh.");
+      }
+
+    /* Remove these once suport for mesh-options/mesh_option is removed */
+    if( mesh_build_type == "create_1D_mesh" /* This is deprecated */ )
+      dimension = 1;
+
+    if( mesh_build_type=="create_2D_mesh" /* This is deprecated */ )
+      dimension = 2;
+
+    if( mesh_build_type=="create_3D_mesh" /* This is deprecated */ )
+      dimension = 3;
+
+    // Set the mesh dimension
+    mesh->set_mesh_dimension(dimension);
+
+    libMesh::Real x_min = input("Mesh/Generation/x_min", 0.0);
+    if( input.have_variable("mesh-options/domain_x1_min") )
+      {
+        std::string warning = "WARNING: mesh-options/domain_x1_min is DEPRECATED.\n";
+          warning += "         Please update to use Mesh/Generation/x_min.\n";
+          grins_warning(warning);
+
+          x_min = input("mesh-options/domain_x1_min", 0.0);
+      }
+
```

Comment:
You've got about a brazillion of these blocks, which really ought to be factored into a single (template) function.

I'd also upgrade from warning to error in the case where both the new and the old option is found in the same config file.  Otherwise I could imagine new-style input file A accidentally overriding itself by including old-style input file B.